src/addEffect.coffee

```diff
@@ -1,43 +1,6 @@
 import {useEffect} from 'react'
-import {isArray, get, shallowEqualArray} from './util/helpers'
-import usePrevious from './util/usePrevious'
+import addSomeEffect from './addSomeEffect'
 
-isSimpleDependenciesArray = (dependencies) ->
-  return no unless isArray dependencies
-  for element in dependencies
-    if element.indexOf('.') > -1
-      return no
-  yes
-
-addEffect = (callback, dependencies) -> (props) ->
-  # eslint-disable-next-line coffee/no-negated-condition
-  if not dependencies?
-    useEffect callback props
-  else if isSimpleDependenciesArray dependencies
-    # TODO: throw nice error if changeProps isn't array/iterable or any changeProp isn't a string?
-    useEffect(
-      callback props
-      (props[dependencyPropName] for dependencyPropName in dependencies)
-    )
-  else
-    prevProps = usePrevious props
-    if isArray dependencies
-      useEffect ->
-        return if (
-          prevProps? and
-          shallowEqualArray(
-            (get(dependencyName) prevProps for dependencyName in dependencies)
-            (get(dependencyName) props for dependencyName in dependencies)
-          )
-        )
-
-        callback(props)()
-    else
-      useEffect ->
-        return if prevProps? and not dependencies prevProps, props
-
-        callback(props)()
-
-  props
+addEffect = addSomeEffect useEffect
```

Comment:
Since `useEffect` and `useLayoutEffect` are documented as having the same signature, and have very similar behavior, moving this code into a generic `addSomeEffect` function (naming?) seems reasonable to me.

Reply:
Good choice sharing code with `addEffect()`, let's just change the name to something that doesn't use the `add*` convention (since it's a helper generator rather than a helper)

Maybe eg `createEffectAdder()`?